test/OpenTelemetry.Tests/Instrumentation/PropertyFetcherTest.cs

```diff
@@ -50,5 +50,35 @@ public void FetchNullProperty()
             var fetch = new PropertyFetcher<string>("null");
             Assert.False(fetch.TryFetch(null, out _));
         }
+
+        [Fact]
+        public void FetchPropertyMultiplePayloadTypes()
+        {
+            var fetch = new PropertyFetcher<string>("Property");
+
+            Assert.True(fetch.TryFetch(new PayloadTypeA(), out string propertyValue));
+            Assert.Equal("A", propertyValue);
+
+            Assert.True(fetch.TryFetch(new PayloadTypeB(), out propertyValue));
+            Assert.Equal("B", propertyValue);
```

Comment:
Interesting, so if most of the fetches are against `PayloadTypeB` but the first invocation for a process is against `PayloadTypeA` there'll always be a fetch first assuming `A` and then another for `B`. Probably not a huge deal overhead-wise, but just explicitly calling it out.

Reply:
Ya if you look at the issue it looks like some kind of Azure Functions thing where it always does a gRPC call first. So for native HttpClient invocations in that environment, it is going to pay the penalty of a failed `is` check and then an extra virtual dispatch on each call. Very minor hit, but a hit nonetheless. Not much we can do about it. If we instead did a dictionary lookup, we would have to pay for a hash + lookup for _all_ invocations even in cases where only one payload is ever used. So this seemed like the lesser evil. Kind of lame this gRPC path uses the same diagnostic source but with different payload types.